### PR TITLE
Allow to set GEANT UserDecay via configurable params

### PR DESCRIPTION
**Pull request by preghenella**
This PR allows the user to switch on Geant user decay for any given particle.
It is technically done via configurable parameters at command line.
As an example
```
--configKeyValues 'SimUserDecay.pdg=3122 111;
```
will turn on user decay for Lambda0 (3122) and pi0 (111) which will be in turn be decayed by the external decayer. 

The feature will be useful for users that need productions with force decays and/or decay chains.
An example of use is given in
```
run/SimExamples/ForceDecay_Lambda_Neutron_Dalitz/
```

The PR also adds the `DecayerPythia8.verbose` parameter to have a verbose logging.

*****
This is the README.md of the sim example.

This is a simple simulation example showing how to force a specific particle decay chain.
To simplify the example, we will use a Box generator to inject Lambda0 particles in the simulation.

What we want to achieve is a forced decay chain of the Lambda0
```
Lambda0 --> n pi0 --> n e+ e- gamma
```
Given that both the Lambda0 and the pi0 are by default decayed by Geant, we have to turn on the external decayer for these particles.
This can be done with
```
--configKeyValues 'SimUserDecay.pdg=3122 111'
```

On top of that we have to setup the external decayer to perform the decay forcing the channels we desire.
The default external decayer configuration is loaded from
```
${O2_ROOT}/share/Generators/pythia8/decays/base.cfg
```
What we want to do is to add on top of the default configuration some extra settings that are provided in the `decay_lambda_neutron_dalitz.cfg` file.
This can be done with
```
--configKeyValues 'DecayerPythia8.config=${O2_ROOT}/share/Generators/pythia8/decays/base.cfg decay_lambda_neutron_dalitz.cfg'
```

### CODE REVIEW

**Comment by sawenzel** [comment]
@preghenella: Since one can give multiple pdgs to `SimUserDecay.pdg`, I suggest to call it `SimUserDecay.pdgs`.

**Comment by preghenella** [comment]
that was the original name I have thought.
I will rename it to `pdglist` to be even clearer

File: Steer/src/O2MCApplication.cxx [hunk 1]
```diff
@@ -172,6 +174,28 @@ void O2MCApplicationBase::BeginEvent()
...
 }
 
+void O2MCApplicationBase::AddParticles()
+{
+  // dispatch first to function in FairRoot
+  FairMCApplication::AddParticles();
+
+  auto& param = o2::conf::SimUserDecay::Instance();
+  LOG(INFO) << "Printing \'SimUserDecay\' parameters";
+  std::cout << param << std::endl;
```
Comment (sawenzel) [inline Steer/src/O2MCApplication.cxx:184]: why do you make the distinction between LOG(INFO) and std::cout here? (It should probably be the same)

Comment (preghenella) [reply Steer/src/O2MCApplication.cxx:184]: it's a mistake. it was supposed to be LOG(INFO).
I will fix that, thanks for catching it.

**Comment by preghenella** [comment]
Dear @ihrivnac @amorsch ,
for the record, the mechanism implemented in this PR to force decays via the external decayer
```
TVirtualMC::GetMC()->SetUserDecay(pdg);
```
works for both G3 and G4.
Roberto

**Pull request merged**